### ISSUE TITLE
no-shadowed-variable: type aliases and variables shouldn't count as shadowing each other

### ISSUE THREAD

**Issue by ethanresnick** [palantir/tslint#3210]
### Bug Report

- __TSLint version__: 5.6.0
- __TypeScript version__: 2.5.1
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
function x() {
  type someName = any;
  if(true) {
    const someName = true;
  }
}
```

#### Actual behavior
no-shadowed-variable: Shadowed name: 'someName'

#### Expected behavior
No error, as no shadowing is occurring here. Also, naming a type after a variable can be useful. For example, when an intermediate result needs to be stored in a variable, naming the type after the variable whose contents it describes makes sense, since the type won't be reused in another context. I've needed to do this when TS can't correctly infer the type of the variable.

**Comment by ajafff** [comment]
This is a duplicate of #3021
I'll try to get this implemented soon

**Comment by ethanresnick** [comment]
Cool, thanks. I saw #3021, but didn't realize that namespaces were runtime values in typescript and not type definitions.
